plugins/notifications-backend/src/service/handlers.ts

```diff
@@ -0,0 +1,425 @@
+    origin: req.origin,
+    title: req.title,
+    message: req.message,
+    topic: req.topic,
+    is_system: !isUser,
+  };
+
+  let messageId: string;
+
+  const ret = dbClient.transaction(trx => {
+    return trx
+      .insert(row)
+      .returning<string, { id: string }[]>('id')
+      .into('messages')
+      .then(ids => {
+        messageId = ids[0].id;
+        if (Array.isArray(req.targetUsers)) {
+          const userInserts = req.targetUsers.map(user => {
+            return {
+              message_id: messageId,
+              user: user,
+            };
+          });
+          return trx('users').insert(userInserts);
+        }
+
+        return undefined;
+      })
+      .then(() => {
+        if (Array.isArray(req.targetGroups)) {
+          const groupInserts = req.targetGroups.map(group => {
+            return {
+              message_id: messageId,
+              group: group,
+            };
+          });
+          return trx('groups').insert(groupInserts);
+        }
+
+        return undefined;
+      })
+      .then(() => {
+        if (Array.isArray(req.actions)) {
+          const actionInserts: ActionsInsert[] = req.actions.map(action => {
+            return {
+              url: action.url,
+              title: action.title,
+              message_id: messageId,
+            };
+          });
+
+          return trx('actions').insert(actionInserts);
+        }
+
+        return undefined;
+      })
+      .then(() => {
+        return { messageId: messageId };
+      });
+  });
+
+  return ret;
+}
+
+// getNotifications
+export async function getNotifications(
+  dbClient: Knex<any, any>,
+  loggedInUser: string,
+  catalogClient: CatalogClient,
+  filter: NotificationsFilterRequest,
+  pageSize: number = DefaultPageSize,
+  pageNumber: number = DefaultPageNumber,
+  sorting: NotificationsSortingRequest,
+): Promise<Paths.GetNotifications.Responses.$200> {
+  if (
+    pageSize < 0 ||
+    pageNumber < 0 ||
+    (pageSize === 0 && pageNumber > 0) ||
+    (pageSize > 0 && pageNumber === 0)
+  ) {
+    throw new Error(
+      'pageSize and pageNumber must both be either 0 or greater than 0',
+    );
+  }
+
+  if (!filter.messageScope) {
+    filter.messageScope = DefaultMessageScope;
+  } else if (!MessageScopes.includes(filter.messageScope)) {
+    throw new Error(
+      `messageScope parameter must be one of ${MessageScopes.join()}`,
+    );
+  }
+
+  const orderBy = sorting.orderBy || DefaultOrderBy;
+  const orderByDirec = sorting.OrderByDirec || DefaultOrderDirection;
+  if (
+    !NotificationsOrderByFields.includes(orderBy) ||
+    !NotificationsOrderByDirections.includes(orderByDirec)
+  ) {
+    throw new Error(
+      `The orderBy parameter can be one of ${NotificationsOrderByFields.join(
+        ',',
+      )}. The orderByDirec can be either ${NotificationsOrderByDirections.join(
+        ' or ',
+      )}.`,
+    );
+  }
+
+  const userGroups = await getUserGroups(catalogClient, loggedInUser);
```

Comment:
You don't actually have to ask this from the catalog but it's included in the `identityApi.getIdentity` response when you are authenticating the user (in `identity.ownershipEntityRefs`)

Reply:
based on the knowledge I have backstage is not really used with authentication (not yet). so we do need it for installations without authentication.